CRM/Civicase/Hook/SelectWhereClause/RestrictSearchToAccessibleCaseCategories.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+use CRM_Civicase_Helper_CaseCategory as CaseCategoryHelper;
+
+/**
+ * Restrict search to accessible case categories if case entity is involved.
+ */
+class CRM_Civicase_Hook_SelectWhereClause_RestrictSearchToAccessibleCaseCategories {
+
+  /**
+   * Restrict search to accessible case categories if case entity is involved.
+   *
+   * @param string $entity
+   *   Entity name.
+   * @param array $clauses
+   *   Where clauses.
+   */
+  public function run($entity, array &$clauses) {
+
+    if (!$this->shouldRun($entity)) {
+      return;
+    }
+    $accessibleCaseCategories = CaseCategoryHelper::getAccessibleCaseTypeCategories();
+    $params = (json_decode($_REQUEST['json'], TRUE));
+
+    if (
+      isset($params[0][2]['case_type_id.case_type_category']) &&
```

Comment:
There are so many uncertainties here. 
- How are we sure that this will always be the position of the `case_type_id.case_type_category ` param in the `params` array?
- The `params` array is gotten from the `$_REQUEST['json']`, will this always be populated when the Case API is called from PHP and from Frontend and also when invoked from searches?